src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs

```diff
@@ -212,104 +280,173 @@ protected override void HandleTaskExecutionCore()
-                    {
-                        if (package.LineVertexCount > 0 && !package.LineStripVertexColors.Any())
-                        {
-                            package.ApplyLineVertexColors(CreateColorByteArrayOfSize(package.LineVertexCount, DefR, DefG, DefB, DefA));
-                        }
+            if (package.LineVertexCount > previousLineVertexCount)
+            {
+                var count = package.LineVertexCount - packageSupplement.LineVertexColorCount;
+                if (count > 0)
+                {
+                    packageSupplement.AppendLineVertexColorRange(CreateColorByteArrayOfSize(count, DefR, DefG, DefB, DefA));
+                }
+            }
 
-                        if (package.PointVertexCount > 0 && !package.PointVertexColors.Any())
-                        {
-                            package.ApplyPointVertexColors(CreateColorByteArrayOfSize(package.PointVertexCount, DefR, DefG, DefB, DefA));
-                        }
-                    }
+            if (package.MeshVertexCount > previousMeshVertexCount)
+            {
+                var count = package.MeshVertexCount - packageSupplement.MeshVertexColorCount;
+                if (count > 0)
+                {
+                    packageSupplement.AppendMeshVertexColorRange(CreateColorByteArrayOfSize(count, DefR, DefG, DefB, DefA));
                 }
-                catch (Exception e)
+            }
+        }
+
+        private static void LegacyPackageEnsureColorExistsPerVertex(IRenderPackage pkg, int previousPointVertexCount, int previousLineVertexCount, int previousMeshVertexCount)
+        {
+            if (pkg.PointVertexCount > previousPointVertexCount
+                && pkg.PointVertexColors.Any()
+                && pkg.PointVertexColors.Count() / 4 > pkg.PointVertexCount)
+            {
+                for (var i = pkg.PointVertexColors.Count() / 4; i < pkg.PointVertexCount; i++)
                 {
-                    Debug.WriteLine(
-                        "PushGraphicItemIntoPackage: " + e);
+                    pkg.AddPointVertexColor(DefR, DefG, DefB, DefA);
                 }
+            }
 
-                package.DisplayLabels = displayLabels;
-                package.IsSelected = isNodeSelected;
+            if (pkg.LineVertexCount > previousLineVertexCount
+                && pkg.LineStripVertexColors.Any()
+                && pkg.LineStripVertexColors.Count() / 4 > pkg.LineVertexCount)
+            {
+                for (var i = pkg.LineStripVertexColors.Count() / 4; i < pkg.LineVertexCount; i++)
+                {
+                    pkg.AddLineStripVertexColor(DefR, DefG, DefB, DefA);
+                }
+            }
 
-                renderPackageCache.Add(package, outputPortId);
+            if (pkg.MeshVertexCount > previousMeshVertexCount
+                && pkg.MeshVertexColors.Any()
+                && pkg.MeshVertexColors.Count() / 4 > pkg.MeshVertexCount)
```

Comment:
Shouldn't this check be `pkg.MeshVertexCount > pkg.MeshVertexColors.Count() / 4`?